cmd/krew/cmd/update.go

```diff
@@ -41,12 +57,147 @@ Remarks:
 	RunE: ensureIndexUpdated,
 }
 
+func retrieveUpdatedPluginList() ([]string, error) {
+	modifiedFiles, err := gitutil.ListModifiedFiles(constants.IndexURI, paths.IndexPluginsPath())
+	if err != nil {
+		return []string{}, err
+	}
+
+	var plugins []string
+	for _, f := range modifiedFiles {
+		filename := filepath.Base(f)
+		extension := filepath.Ext(filename)
+		name := filename[0 : len(filename)-len(extension)]
+		plugins = append(plugins, name)
+	}
+
+	return plugins, nil
+}
+
+func retrievePluginNameVersionMap(names []string) map[string]string {
+	m := make(map[string]string, len(names))
+	for _, n := range names {
+		plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(), n)
+		if err != nil {
+			continue
+		}
+
+		m[n] = plugin.Spec.Version
+	}
+
+	return m
+}
+
+func retrieveInstalledPluginMap() (map[string]bool, error) {
+	plugins, err := installation.ListInstalledPlugins(paths.InstallReceiptsPath())
+	if err != nil {
+		return map[string]bool{}, err
+	}
+	m := make(map[string]bool, len(plugins))
+	for name := range plugins {
+		m[name] = true
+	}
+
+	return m, nil
+}
+
+func filterAndSortUpdatedPlugins(old, updated map[string]string, installed map[string]bool) ([]newPlugin, []newVersionPlugin) {
+	var newPluginList []newPlugin
+	var newVersionList []newVersionPlugin
+
+	for name, version := range updated {
+		oldVersion, ok := old[name]
+		if !ok {
+			newPluginList = append(newPluginList, newPlugin{
+				name:    name,
+				version: version,
+			})
+			continue
+		}
+
+		if version != oldVersion {
+			_, installed := installed[name]
+			newVersionList = append(newVersionList, newVersionPlugin{
+				name:       name,
+				installed:  installed,
+				oldVersion: oldVersion,
+				newVersion: version,
+			})
+			continue
+		}
+	}
+
+	sort.Slice(newPluginList, func(i, j int) bool {
+		return newPluginList[i].name < newPluginList[j].name
+	})
+
+	sort.Slice(newVersionList, func(i, j int) bool {
+		if newVersionList[i].installed && !newVersionList[j].installed {
+			return true
+		}
+
+		if !newVersionList[i].installed && newVersionList[j].installed {
+			return false
+		}
+
+		return newVersionList[i].name < newVersionList[j].name
+	})
+
+	return newPluginList, newVersionList
+}
+
+func showUpdatedPlugins(newPluginList []newPlugin, newVersionList []newVersionPlugin) {
+	if len(newPluginList) > 0 {
+		fmt.Fprintln(os.Stderr, "  New plugins available:")
+		for _, np := range newPluginList {
+			fmt.Fprintf(os.Stderr, "    * %s %s\n", np.name, np.version)
```

Comment:
I think a `TabWriter` would produce more legible output here.

But as far as I understand, a one-line report would be preferable anyways.